rest_framework_json_api/views.py

```diff
@@ -98,6 +100,54 @@ def get_queryset(self, *args, **kwargs):
         return qs
 
 
+class RelatedMixin(object):
+    """
+    This mixin handles all related entities, whose Serializers are declared in "related_serializers"
+    """
+    related_serializers = {}
+    field_name_mapping = {}
```

Comment:
what's the use case that this mapping is needed?

Reply:
That is used when url `related_field` name differs from model field name. Let's say we have field `Author.bio`,  but url we want is `/author/1/biography/`. Then we do:
```python
field_name_mapping = {'biography': 'bio'}
```

Reply:
Is this not the same which can already be done in the serializer with `source` attribute on a field? Would it be not more consistent to take the mapping information form the serializer directly?

Reply:
Just tried. No. We trying to get related_attribute from **model**, but not from serializer, so it won't work. I can remove it, but there will be no opportunity to rename relation in the url.

Reply:
Are you sure? I assume for consistency the url relation part should have the same name as the relation in the serializer right?

What we could to ensure this is get the field declaration from the serializer class, read the source attribute and use this to get the relation from the model we have access to in case it is set. This should work or not?

Reply:
>Are you sure? I assume for consistency the url relation part should have the same name as the relation in the serializer right?

100% agree with you. But there are some cases when we need to have different name. For example Author.entry_set. It would be nicer to have url like /author/1/entries/, right ?

>What we could to ensure this is get the field declaration from the serializer class, read the source attribute and use this to get the relation from the model we have access to in case it is set. This should work or not?

I [tried](https://github.com/Anton-Shutik/django-rest-framework-json-api/commit/c6a77654873fa5d2460c43bc6b7efa4474ccdb46) (If you know better/nicer way to do it - pls let me know ). Looks like we have to create Serializer instance to be able to get the field. That looks more complex, but seems like it works. 
What do we do when serializer does not have such a field ? Should we just return 500 or try to use related_name we got from the url ? Will it be possible to have a related link without declaring the field on the serializer ? You probably will want to deny such a case, but seems like json api spec allows it, or at least does not deny.